### PR TITLE
Horizontal progress bar implimentation

### PR DESCRIPTION
**Pull request by amuwhao46**
Finished implimenting the bar. saving the document in main automatically formats it. I didn't change the actual code

### CODE REVIEW

File: .dart_tool/package_config_subset [hunk 1]
```diff
@@ -54,6 +54,10 @@ path
 2.12
 file:///Users/thenamesoak/.pub-cache/hosted/pub.dev/path-1.8.2/
```
Comment (Rey-Jey) [inline .dart_tool/package_config_subset:55]: These package locations only work on your local machine. We need to figure out how to solve this.

**Review by Rey-Jey — COMMENTED** [review]
Code looks good, issue resolved

**Review by JayanthAnandNama — APPROVED** [review]
Code looks good!

**Review by Ahmed-Alagbari — APPROVED** [review]
progress bar looks good!

**Review by Rey-Jey — APPROVED** [review]
Good work

**Pull request merged**